mesonbuild/interpreterbase.py

```diff
@@ -926,6 +928,22 @@ def evaluate_ternary(self, node: mparser.TernaryNode) -> TYPE_var:
         else:
             return self.evaluate_statement(node.falseblock)
 
+    def evaluate_fstring(self, node: mparser.FormatStringNode) -> TYPE_var:
+        assert(isinstance(node, mparser.FormatStringNode))
+
+        def replace(match: T.Match[str]) -> str:
+            var = str(match.group(1))
+            try:
+                val = self.variables[var]
+                if not isinstance(val, (str, int, float, bool)):
+                    raise InvalidCode(f'Identifier "{var}" does not name a formattable variable.')
```

Comment:
Maybe add here that it must be one of string, integer, float or boolean, to give a hint to user what they can do.